src/qobj_v2.jl

```diff
@@ -0,0 +1,69 @@
+function yaotoqobjv2(qc::Array{ChainBlock{N}}, device::String; nshots=1024) where N
+    nslots = 1
+    main_header = Dict("description"=>"Set of Experiments 1", "backend_name" => "$(device)")
+    main_config = Dict("shots"=>nshots, "memory_slots"=>nslots, "init_qubits"=> true)
+    experiments = collect(generate_experiment(i) for i in qc)
+    data = Dict("qobj_id" => "foo", "schema_version"=>"1.0.0", "type"=>"QASM", "header"=>main_header, "config"=>main_config, "experiments"=>experiments)
+    Qobj(data)
+end
+
+function generate_experiment(qc::ChainBlock)
+    qc_simpl = basicstyle(qc)
+    n_qubits = nqubits(qc_simpl)
+    n_classical_reg = 2 # figure out this info
+    nslots=1
+    c_label = [["c", i] for i in 0:n_classical_reg-1]
+    q_label = [["q", i] for i in 0:n_qubits-1]
+    exp_inst = generate_instv2(qc_simpl)
+    exp_header = Dict("memory_slots"=>nslots, "n_qubits"=>n_qubits, "clbit_labels"=>c_label, "qubit_labels"=>q_label)
+    experiment = Dict("header"=>exp_header, "config"=>Dict(), "instructions"=>exp_inst)
+    return experiment    
+end
+
+function generate_instv2(qc_simpl::ChainBlock)
+    inst = []
+    for block in subblocks(qc_simpl)
+        # push!(inst, generate_instv2(block))
+        if block isa ChainBlock
+            push!(inst, generate_instv2.(block)...)
+        else
+            push!(inst, generate_instv2(block))
+        end
+    end
+    return inst
+end
+
+function generate_instv2(blk::PutBlock{N,M}) where {N,M}
+	locs = [blk.locs...]
+	generate_instv2(blk.content, locs)
+end
+
+function generate_instv2(blk::ControlBlock{N,GT,C}) where {N,GT,C}
```

Comment:
```julia
julia> c = control(5, -3, 2=>X)
nqubits: 5
control(¬3)
└─ (2,) X

julia> IBMQJulia.generate_instv2(c)
Dict{String,Any} with 2 entries:
  "name"   => "cx"
  "qubits" => [2, 3]
```

Please check all variables in `ctrl_config` are `1`, and error on any `0`. The `0` location represents the inverse control.